### DISCUSSION TITLE
Container doesn't contain curl or wget needed to use healthcheck on Fargate

### DISCUSSION THREAD

**Discussion by NeilJed** [signalfx/splunk-otel-collector#1955] [Ideas]
The default Fargate configuration enables the healthcheck endpoint and I want to use this as part of my task definition to ensure the Otel Collector sidecar is in service.

However, as the checkis run internally from within the container, this can't be done as it doesn't ship with curl, wget or another shell tool able to make a query to the endpoint and return a status code.

I think the container needs one of these two to enable healtchecks, at least on Fargate, to work.

**Comment by rmfitzpatrick** [comment]
Hi, curl is available in the agent bundle at `/usr/lib/splunk-otel-collector/agent-bundle/bin/curl` though this isn't a guaranteed stable location (you will need to set the entrypoint to bash). It may be preferable for you to use the collector image as a partial base and provide your own utilities in a custom image.

**Comment by NeilJed** [comment]
TBH I'd rather avoid having to build and maintain our own custom image just to add curl to execute a healthcheck ☹️

How difficult would it be to make the curl location stable at least? I think providing a reliable means of performing a healthcheck on the container seems like a fairly important feature?

**Comment by NeilJed** [comment]
https://github.com/signalfx/splunk-otel-collector/pull/4646

So I guess container based healtchecks are dead then?

**Reply by harshitrjpt** [reply]
Python is available (atleast for now)

try
```
python -c "import requests; exit(1) if not \"Server available\" in requests.get('http://localhost:13133').text else exit(0)"
```
